src/test/java/com/gmail/woosay333/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,337 @@
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isAccepted())
+                .andReturn();
+
+        BookResponseDto actual = objectMapper
+                .readValue(result.getResponse().getContentAsString(), BookResponseDto.class);
+        assertNotNull(actual);
+        assertNotNull(actual.id());
+        assertEquals(VALID_BOOK_ID_HOBBIT, actual.id());
+        assertTrue(EqualsBuilder.reflectionEquals(expected, actual, "id"));
+    }
+
+    @Test
+    @DisplayName("""
+            Get existing book by ID,
+            expected status - 200, expected response - BookResponseDto
+            """)
+    @WithMockUser
+    void getBookById_ExistingBookId_ReturnBookResponseDto() throws Exception {
+        BookResponseDto expected = BookResponseDto.builder()
+                .id(VALID_BOOK_ID_HOBBIT)
+                .author(VALID_BOOK_AUTHOR_TOLKIEN)
+                .title(VALID_BOOK_TITLE_HOBBIT)
+                .categoryIds(VALID_BOOK_CATEGORY_IDS_HOBBIT)
+                .isbn(VALID_BOOK_ISBN_HOBBIT)
+                .price(VALID_BOOK_PRICE_HOBBIT)
+                .build();
+        String url = URI + "/" + VALID_BOOK_ID_HOBBIT;
+
+        MvcResult result = mockMvc.perform(get(url)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookResponseDto actual = objectMapper
+                .readValue(result.getResponse().getContentAsString(), BookResponseDto.class);
+        assertNotNull(actual);
+        assertNotNull(actual.id());
+        assertTrue(EqualsBuilder.reflectionEquals(expected, actual));
+    }
+
+    @Test
+    @DisplayName("""
+            Get three existing books,
+            expected status - 200, expected response - BookResponseDto[]
+            """)
+    @WithMockUser
+    void getAll_ReturnTwoBookResponseDtoAtFirstPageSizeTwoSortByIdAsc() throws Exception {
+        BookResponseDto responseDtoHarryPotter = BookResponseDto.builder()
+                .id(VALID_BOOK_ID_HARRY_POTTER_CHAMBER_OF_SECRETS)
+                .title(VALID_BOOK_TITLE_HARRY_POTTER_CHAMBER_OF_SECRETS)
+                .author(VALID_BOOK_AUTHOR_ROWLING)
+                .categoryIds(VALID_BOOK_CATEGORY_IDS_HARRY_POTTER_CHAMBER_OF_SECRETS)
+                .isbn(VALID_BOOK_ISBN_HARRY_POTTER_CHAMBER_OF_SECRETS)
+                .price(VALID_BOOK_PRICE_HARRY_POTTER_CHAMBER_OF_SECRETS)
+                .build();
+        BookResponseDto responseDtoHobbit = BookResponseDto.builder()
+                .id(VALID_BOOK_ID_HOBBIT)
+                .title(VALID_BOOK_TITLE_HOBBIT)
+                .author(VALID_BOOK_AUTHOR_TOLKIEN)
+                .categoryIds(VALID_BOOK_CATEGORY_IDS_HOBBIT)
+                .isbn(VALID_BOOK_ISBN_HOBBIT)
+                .price(VALID_BOOK_PRICE_HOBBIT)
+                .build();
```

Comment:
maybe we can also move those objects initialization outside of test method? and reuse them, because hobbit for example is used in other tests as well. It can be just a class variable like `hobbitBookResponse` or something like that and we can initialize them in some `beforeEach`, or it can be a private static method that returns that book, maybe you'd even want to put this method in a `TestData` class to not litter the space here